org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationPatching.cs

```diff
@@ -0,0 +1,442 @@
+                        DateTime date;
+
+                        // Get person property or attribute
+                        if ( propertyOrAttribute == "Property" ) // is property
+                        {
+                            value = person.GetPropertyValue(key).ToStringSafe();
+                        }
+                        else // is attribute
+                        {
+                            var attributeQry = attributeService.Queryable().Where( a => a.EntityTypeId == 15 && a.Key == key ).AsNoTracking();
+                            try
+                            {
+                                Debug.WriteLine( "Type/Key: " + " " + type + " / " + hsKey );
+                                value = attributeValueService.Queryable().Where( av => av.EntityId == current_id ).Join( attributeQry, av => av.AttributeId, a => a.Id, ( av, a ) => av ).Select( av => av.Value ).AsNoTracking().FirstOrDefault().ToStringSafe();
+                            }
+                            catch
+                            {
+                                value = "";
+                            }
+                        }
+
+                        // Set date values to HubSpot required format
+                        if ( type == "Date" )
+                        {
+                            if ( string.IsNullOrEmpty(value) == false )
+                            {
+                                if ( value.AsDateTime() != null )
+                                {
+                                    value = ConvertDate( value.AsDateTime() );
+                                }
+                            }
+                        }
+
+                        // Patch it!
+                        Debug.WriteLine( "Patching: " + hsKey + " " + value );
+                        properties.Add( new HubspotPropertyUpdate() { property = hsKey, value = value } );
+
+                    }
+                    
+                    // Handle name changes
+                    properties.Add(new HubspotPropertyUpdate() { property = "firstname", value = person.NickName });
+                    properties.Add(new HubspotPropertyUpdate() { property = "lastname", value = person.LastName });
+
+                    // Handle Phone
+                    PhoneNumber mobile = person.PhoneNumbers.FirstOrDefault( n => n.NumberTypeValueId == 12 );
+                    if ( mobile != null && !mobile.IsUnlisted && mobile.IsMessagingEnabled )
+                    {
+                        properties.Add( new HubspotPropertyUpdate() { property = "phone", value = mobile.NumberFormatted } );
+                    }
+                    else
+                    {
+                        properties.Add( new HubspotPropertyUpdate() { property = "phone", value = "" } );
+                    }
+
+                    // Handle Email
+                    string email = person.Email == "" ? "" : person.Email.ToStringSafe();
```

Comment:
This line is a bit overcomplicated and has some redundant parts. It could be simplified as:
```csharp
string email = person.Email ?? "";
```

The original code says "Set email to the person's email if it is not blank. If the person's email is blank, set the email to blank instead of the person's email". The new code says "Set the email to the person's email, but set it to blank if it is null".